test/prism_regression/class_missing_name.rb

```diff
@@ -0,0 +1,17 @@
+# typed: false
+# disable-parser-comparison: true
+
+# Missing constant path
+class
+end
+
+# Invalid constant path without body
+class
+  def foo; end # Parsed as constant path
+end
+
+# Invalid constant path with body
+class
+  def foo; end # Parsed as constant path
+  def bar; end # Parsed as body
```

Comment:
It differs from the legacy parser, but I think we can make this even by better by not discarding `def foo`, but instead making it the start of the body